app/src/main/java/org/oppia/app/topic/play/ChapterSummaryAdapter.kt

```diff
@@ -0,0 +1,48 @@
+package org.oppia.app.topic.play
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import org.oppia.app.R
+import androidx.databinding.DataBindingUtil
+import android.view.LayoutInflater
+import androidx.databinding.library.baseAdapters.BR
+import org.oppia.app.databinding.PlayChapterViewBinding
+import org.oppia.app.model.ChapterPlayState
+import org.oppia.app.model.ChapterSummary
+
+// TODO(#216): Make use of generic data-binding-enabled RecyclerView adapter.
+
+/** Adapter to bind ChapterSummary to [RecyclerView] inside [TopicPlayFragment]. */
+class ChapterSummaryAdapter(
+  private val chapterList: List<ChapterSummary>,
+  private val chapterSummarySelector: ChapterSummarySelector
+) :
+  RecyclerView.Adapter<ChapterSummaryAdapter.ChapterSummaryViewHolder>() {
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChapterSummaryViewHolder {
+    val chapterSummaryListItemBinding = DataBindingUtil.inflate<PlayChapterViewBinding>(
+      LayoutInflater.from(parent.context),
+      R.layout.play_chapter_view, parent,
+      /* attachToRoot= */ false
+    )
+    return ChapterSummaryViewHolder(chapterSummaryListItemBinding)
+  }
+
+  override fun getItemCount(): Int {
+    return chapterList.size
+  }
+
+  override fun onBindViewHolder(chapterSummaryViewHolder: ChapterSummaryViewHolder, position: Int) {
+    chapterSummaryViewHolder.bind(chapterList[position], position)
+  }
+
+  inner class ChapterSummaryViewHolder(private val binding: PlayChapterViewBinding) :
+    RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(chapterSummary: ChapterSummary, @Suppress("UNUSED_PARAMETER") position: Int) {
```

Comment:
Please remove unused parameter

Reply:
This is an important parameter which can be used in future codes, so its a good practice to pass this parameter and suppress it.

Reply:
It's easy to pipe this data along though, so I personally lean more toward removing the parameter to avoid the code smell of introducing unused code.